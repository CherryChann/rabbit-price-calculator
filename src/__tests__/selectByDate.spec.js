import React from 'react';
import { shallow, mount } from 'enzyme';
import SelectByDate from '../components/selectByDate';

const onChange = () => { console.log('Mock Function')}
const props = {
    startDate: new Date(),
    minDate: new Date(),
    maxDate: new Date(),
    onChange: onChange
}

describe('Shallow rendered select by date component', () => {
    const container = shallow(
        <SelectByDate 
            startDate={props.startDate}
            minDate={props.minDate}
            maxDate={props.maxDate}
            onChange={props.onChange}
        />);
    it('should render date picker', () => {
        expect(container.html()).toMatchSnapshot();
    });
    it('should have correct props, called props function and fire selected date', () => {
        expect(container.find('.custom-date-picker').length).toEqual(1);
        expect(container.find('.custom-date-picker').prop('minDate')).toEqual(props.minDate);
        expect(container.find('.custom-date-picker').prop('maxDate')).toEqual(props.maxDate);
        expect(container.find('.custom-date-picker').prop('selected')).toEqual(props.startDate);
        container.find('.custom-date-picker').simulate('change', {
            target: {
                value: props.maxDate
            }
        })
        expect(container.find('.custom-date-picker').prop('selected')).toEqual(props.maxDate);
    })
});