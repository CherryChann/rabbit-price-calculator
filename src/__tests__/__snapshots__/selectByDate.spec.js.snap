// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered date  should render date picker 1`] = `"<div class=\\"row\\"><div class=\\"col-lg-2 col-12\\"><span>Date: </span></div><div class=\\"form-group col-lg-3 col-12\\"><div class=\\"react-datepicker-wrapper\\"><div class=\\"react-datepicker__input-container\\"><input type=\\"text\\" value=\\"05-07-2020\\" placeholder=\\"date selection\\" class=\\"custom-date-picker\\"/></div></div></div></div>"`;
